x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutDatafeedAction.java

```diff
@@ -124,33 +136,74 @@ private void handlePrivsResponse(String username, PutDatafeedAction.Request requ
     private void putDatafeed(PutDatafeedAction.Request request, Map<String, String> headers,
                              ActionListener<PutDatafeedAction.Response> listener) {
 
-        clusterService.submitStateUpdateTask(
-                "put-datafeed-" + request.getDatafeed().getId(),
-                new AckedClusterStateUpdateTask<PutDatafeedAction.Response>(request, listener) {
+        String datafeedId = request.getDatafeed().getId();
+        String jobId = request.getDatafeed().getJobId();
+        ElasticsearchException validationError = checkConfigsAreNotDefinedInClusterState(datafeedId, jobId);
```

Comment:
Just an idea that I think makes this more readable, but it's only my opinion. I would refactor the method to return a boolean `areDefinedInClusterState`, then call `if areDefinedInClusterState(...) { listener.onFailure(...); return }`. This way we don't have to store the error in a local variable, etc.

Reply:
`checkConfigsAreNotDefinedInClusterState` can return different exceptions depending on the error case so the refactor isn't simple